handlers/data_access.js

```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
     })
   }
 }
-
-// TODO @Georgii implement
+// GET Single weather station
 function getSingleWeatherStation (stationCatalog) {
-  return async function (c, req, res, next) {}
+  return async function (c, req, res, next) {
+
+    function getUrlForMeasuredValuesOrForecast (station, req, parameter) {
+      return url.format({
+        protocol: req.protocol,
+        host: req.get('host'),
+        pathname: 'weather-stations/' + station.stationId + '/' + parameter
+      })
+    }
+
+    const stations = su.findStationsInVicinityOf(stationCatalog, undefined, undefined, undefined)
+    const urlString = req.originalUrl
+    const stationId = urlString.split('/')[2]
```

Comment:
all variables should be declared at the top of a function -- javascript does this anyway ("hoisting"), improves readability compared to mix of variable- and function declarations here